### PR TITLE
feat: Introduce compatibility with native namespace packages

### LINKED ISSUE #313: use native support for namespace packages
Per: https://github.com/googleapis/google-auth-library-python/issues/595#issuecomment-683903062

> It looks like different Google API packages have different approaches to using or not using pkg_resources here; e.g. https://github.com/googleapis/python-cloudbuild uses native NS packages and https://github.com/googleapis/python-bigquery uses pkg_resources with a pkgutil fallback.

Now that Python 2.7 and 3.5 support has been dropped, we may be able to follow the example in python-cloudbuild

### PR DESCRIPTION
**Pull request by parthea**
Similar to https://github.com/googleapis/google-auth-library-python/pull/1205
Also see https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#native-namespace-packages

Fixes #313 🦕

### CODE REVIEW

**Comment by parthea** [comment]
Adding `do not merge` as there is an upcoming release freeze. This should be merged after the release freeze ends on November 29th.

File: tests/unit/test_packaging.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+def test_namespace_package_compat(tmp_path):
+    # The ``google`` namespace package should not be masked
+    # by the presence of ``google-cloud-bigquery``.
+    google = tmp_path / "google"
+    google.mkdir()
+    google.joinpath("othermod.py").write_text("")
+    env = dict(os.environ, PYTHONPATH=str(tmp_path))
+    cmd = [sys.executable, "-m", "google.othermod"]
+    subprocess.check_call(cmd, env=env)
```
Comment (kiraksi) [inline tests/unit/test_packaging.py:28]: Were you able to run this test with subprocess without specifying to do it in the shell? It wasn't able to find the tmp path without me specifying to run in the shell, which is the small edit I did in the other PR for this

**Review by kiraksi — APPROVED** [review]
LGTM

**Comment by parthea** [comment]
Superseded by https://github.com/googleapis/python-bigquery/pull/1726

**Pull request closed without merging**